### DISCUSSION TITLE
How to handle the spec file in git

### DISCUSSION THREAD

**Discussion by FilippoBonazziSUSE** [openSUSE/openSUSEway#101] [Ideas]
Following up on the discussion in #74.

Since I submitted a `_service` file using `obs_scm` wth an `obscpio` archive rather than a `tar.gz`, the [obs-service-extract_file](https://github.com/openSUSE/obs-service-extract_file) service does not work.

However, once you run the services with `osc service mr` you have the Git repo checked out right there, in the working folder, with the HEAD detached on the release tag. So the most straightforward way to obtain the spec file is `cp openSUSEway/openSUSEway.spec .` right after `osc service mr`. But I can't find a way to have this dumb simple operation be automated by a service...

**Comment by FilippoBonazziSUSE** [comment]
I just found the `extract` parameter of the `obs_scm` service, which sounds exactly like what we need here. The things you find out when you RTFM...

I will try it and submit a SR if it works.

**Comment by FilippoBonazziSUSE** [comment]
It works fine, submitted in sr#1063017.

I've also added the `format_spec_file` service. This will format the spec file after it's been extracted from the `obscpio` archive. Any meaningful changes to the spec file applied by this service need to be manually submitted to the upstream Git repo.

**Comment by denisok** [comment]
Thanks!
